mainapp/helpers.py

```diff
@@ -186,10 +187,34 @@ def combine_ids(cv_bookids, embedding_bookids, already_rated, recommendations=9)
 
     if len(best_bookids) < recommendations:
         # If not enough recommendations
-        best_bookids = best_bookids + cv_bookids[3:(recommendations + 3 - len(best_bookids))]
+        two_n = (recommendations - len(best_bookids))
+        n1, n2 = math.ceil(two_n/2), math.floor(two_n/2)
+        best_bookids_remain = cv_bookids[3: 6+n1]
+        best_bookids_remain = list(set(best_bookids_remain).difference(set(best_bookids)))[:n1]
+        genre_recomm_bookids = most_common_genre_recommendations(best_bookids, already_rated, best_bookids_remain, n2)
+        best_bookids = best_bookids + best_bookids_remain + genre_recomm_bookids
     return best_bookids
 
 
+def most_common_genre_recommendations(best_bookids, already_rated, best_bookids_remain, n):
+    '''
+        Returns n top rated of the most_common_genre among all lists taken as input
+    '''
+    # Final list of bookids to be recommended
+    books = set(best_bookids+list(already_rated))
```

Comment:
You need to find common genre among -  `best_bookids`,`already_rated` & the first half which will be recommended